### ISSUE TITLE
BackoffSupervisor.withReplyWhileStopped may not respond to messages sent while child actor is just shutting down

### ISSUE THREAD

**Issue by jrudolph** [akka/akka#27000]
That seems to be a race condition between forwarding messages to the child actor and that child actor shutting down. The replyWhileStopped will only be sent after the supervisor has seen the `Terminated` message which may take a while.

Not sure if we can or want to do something about it. It seems the general race condition between forwarding a message and observing `Terminated` of the child (or alternatively observing it in the supervision strategy) cannot be solved.

**Comment by patriknw** [comment]
I agree that this can't be solved. Should be documented, and given at-most-once delivery dropping is ok (although not desired). Alternative would be to deprecate and remove the `withReplyWhileStopped` feature.

**Comment by helena** [comment]
Can we have the supervisor stash or retry if configured to restart, vs DeadLetter if the user choses?

**Comment by chbatey** [comment]
> Can we have the supervisor stash or retry if configured to restart, vs DeadLetter if the user choses?

It wouldn't know that it has to stash as it is a race between the actor stopping and the supervisor receiving the Terminated. A retry would require a protocol with an Ack.
